components/fxa-client/ffi/src/lib.rs

```diff
@@ -198,19 +199,26 @@ pub extern "C" fn fxa_get_manage_devices_url(
 /// A destructor [fxa_str_free] is provided for releasing the memory for this
 /// pointer type.
 #[no_mangle]
-pub extern "C" fn fxa_begin_pairing_flow(
+pub unsafe extern "C" fn fxa_begin_pairing_flow(
     handle: u64,
     pairing_url: FfiStr<'_>,
     scope: FfiStr<'_>,
     entrypoint: FfiStr<'_>,
+    metrics_params: *const u8,
+    metrics_params_len: i32,
     error: &mut ExternError,
 ) -> *mut c_char {
     log::debug!("fxa_begin_pairing_flow");
     ACCOUNTS.call_with_result_mut(error, handle, |fxa| {
         let pairing_url = pairing_url.as_str();
         let scope = scope.as_str();
         let scopes: Vec<&str> = scope.split(' ').collect();
-        fxa.begin_pairing_flow(&pairing_url, &scopes, entrypoint.as_str())
+        let metrics_params =
+            match MetricsParams::from_protobuf_ptr(metrics_params, metrics_params_len) {
```

Comment:
Wasn't sure if I should just error out if something goes wrong with the metrics. Thought I'd just keep it as an option, but I could see this coming back to bite us, since if something breaks in the metrics, we won't be able to see it, unless we explicitly check the url....

Reply:
I think we should error out here - anything that goes wrong at this point will be something that went wrong inside the protobuf library, which we probably want to have show up in sentry.